src/tools/rbd_mirror/PoolReplayer.h

```diff
@@ -82,135 +75,108 @@ class PoolReplayer {
    *  INIT
    *    |
    *    v
-   * <follower> <-------------------------\
-   *    .                                 |
-   *    .                                 |
-   *    v (leader acquired)               |
-   * INIT_IMAGE_MAP             SHUT_DOWN_IMAGE_MAP
-   *    |                                 ^
-   *    v                                 |
-   * INIT_LOCAL_POOL_WATCHER    WAIT_FOR_NOTIFICATIONS
-   *    |                                 ^
-   *    v                                 |
-   * INIT_REMOTE_POOL_WATCHER   SHUT_DOWN_POOL_WATCHERS
-   *    |                                 ^
-   *    v                                 |
-   * INIT_IMAGE_DELETER         SHUT_DOWN_IMAGE_DELETER
-   *    |                                 ^
-   *    v                                 .
-   * <leader> <-----------\               .
-   *    .                 |               .
-   *    . (image update)  |               .
-   *    . . > NOTIFY_INSTANCE_WATCHER     .
-   *    .                                 .
-   *    . (leader lost / shut down)       .
-   *    . . . . . . . . . . . . . . . . . .
+   * <follower> <---------------------\
+   *    .                             |
+   *    . (leader acquired)           |
+   *    v                             |
+   *  NOTIFY_NAMESPACE_WATCHERS     NOTIFY_NAMESPACE_WATCHERS
+   *    |                             ^
+   *    v                             .
+   * <leader>                         .
+   *    .                             .
+   *    . (leader lost / shut down)   .
+   *    . . . . . . . . . . . . . . . .
    *
    * @endverbatim
    */
 
-  typedef std::vector<std::string> InstanceIds;
```

Comment:
Nit: why did you drop the typedef and replace w/ a more verbose `std::vector<std::string>` everywhere?

Reply:
Because it does not hide `std::vector<std::string>` well here. The typedef is in `PoolReplayer.h` but is used for `instance_ids` list, which is passed through methods from different classes where the parameter is either passed as `std::vector<std::string>` (`NamespaceReplayer::handle_instances_added/removed` ,`ImageMap::update_instances_added/removed`,  `LeaderWatcher::list_instances`) or as `leader_watcher::InstanceIds`. If one wanted to change it e.g. to `std::list<std::string>` it would not be just one place. I suppose a more cleaner solution would be to define `InstanceIds` in rbd_mirror/Types.h and update all needed methods, but for me just using `std::vector<std::string>` looked good enough.